web/src/hooks/usePoiFeatures.ts

```diff
@@ -0,0 +1,98 @@
+import { Map } from 'maplibre-gl'
+import { useCallback, useEffect, useMemo } from 'react'
+import { useNavigate, useSearchParams } from 'react-router-dom'
+
+import { GeoJsonPoi, isMultipoi, PoiFeature, PoiModel } from 'api-client'
+
+const multipoiKey = 'multipoi'
+
+const usePoiFeatures = (
```

Comment:
Optional: I struggled a bit with the responsibilities between the two hooks, PoisPage and PoisMobile/Desktop. I created a proposal which might make things a little more clearer: #956
Feel free to reject.